opal/include/opal/sys/powerpc/atomic.h

```diff
@@ -165,31 +165,35 @@ static inline bool opal_atomic_compare_exchange_strong_32 (volatile int32_t *add
     return ret;
 }
 
-static inline int32_t opal_atomic_ll_32 (volatile int32_t *addr)
-{
-   int32_t ret;
-
-   __asm__ __volatile__ ("lwarx   %0, 0, %1  \n\t"
-                         : "=&r" (ret)
-                         : "r" (addr)
-                         );
-   return ret;
-}
-
-static inline int opal_atomic_sc_32 (volatile int32_t *addr, int32_t newval)
-{
-    int32_t ret, foo;
-
-    __asm__ __volatile__ ("   stwcx.  %4, 0, %3  \n\t"
-                          "   li      %0,0       \n\t"
-                          "   bne-    1f         \n\t"
-                          "   ori     %0,%0,1    \n\t"
-                          "1:"
-                          : "=r" (ret), "=m" (*addr), "=r" (foo)
-                          : "r" (addr), "r" (newval)
-                          : "cc", "memory");
-    return ret;
-}
+/* NTH: the LL/SC support is done through macros due to issues with non-optimized builds. The reason
+ * is that even with an always_inline attribute the compiler may still emit instructions to store then
+ * load the arguments to/from the stack. This sequence may cause the ll reservation to be cancelled. */
+#define opal_atomic_ll_32(addr, ret)                                    \
```

Comment:
Makes sense. This is useful comment that should be present in some upper layer. It true for Power and Arm

Reply:
Yeah. I plan to add some additional documentation detailing what live-lock is with LL/SC.